step1-clean/imageProcessor.py

```diff
@@ -0,0 +1,81 @@
+'''
+By: Fitz
+Created: 11 March 2020
+Updated: 12 March 2020
+Total Time to Complete: 7-10
+- 3-3.5 hours  on 11 March 2020
+- 3-3.5 hours on 12 March 2020
+- 2-3 hours wasted on trying to learn deeper unrelated tasks
+
+Image Processing Script
+Related scripts
+- db.py - database setup, config, connect
+- query.py - methods to query database
+
+
+Find and select image processing library for python
+- pillow or pgmagik
+
+Resources
+- https://opensource.com/article/19/3/python-image-manipulation-tools
+- https://pillow.readthedocs.io/en/stable/handbook/index.html
+- https://docs.python.org/3/library/os.html#os-file-dir
+- https://docs.python.org/3/library/sys.html
+- https://docs.sqlalchemy.org/en/13/orm/tutorial.html
+
+'''
+
+import os, sys
+from PIL import Image
+from db import session, SQLImage
+
+def process_image():
+  '''
+    takes user input, create thumbnail, and store in DB
+  '''
+
+  uploadsDirExists = os.path.isdir(os.getcwd() + '/uploads')
```

Comment:
`os.getcwd()` is dangerous here because it returns where the code is being executed from instead of where the code is. For path stuff use `pathlib`: https://stackoverflow.com/questions/3430372/how-do-i-get-the-full-path-of-the-current-files-directory

![Screenshot 2020-03-13 05:39:14](https://user-images.githubusercontent.com/1236436/76609170-4f60eb80-64ed-11ea-880d-4ac589e47162.png)

Reply:
Understood, that is dangerous. I also think pathlib is easier to use once you understand it.